client/src/auth/sign-up.tsx

```diff
@@ -117,106 +106,106 @@ export default function SignUpApp() {
         </div>
       )}
 
-      <DisplaySignupProvider provider={provider || ""} />
-      <DisplayUserDetails details={searchParams.get("user_details") || ""} />
-
-      <label htmlFor="id_terms">
-        <input
-          id="id_terms"
-          type="checkbox"
-          name="terms"
-          checked={checkedTerms}
-          onChange={(event) => {
-            setCheckedTerms(event.target.checked);
-          }}
-        />{" "}
-        I agree to Mozilla's{" "}
-        <a
-          href="https://www.mozilla.org/about/legal/terms/mozilla"
-          target="_blank"
-          rel="noreferrer noopener"
-        >
-          Terms
-        </a>{" "}
-        and{" "}
-        <a
-          href="https://www.mozilla.org/privacy/websites/"
-          target="_blank"
-          rel="noreferrer noopener"
-        >
-          Privacy Notice
-        </a>
-        .
-      </label>
+      <form
+        className="complete-sign-in"
+        method="post"
+        onSubmit={(event) => {
+          event.preventDefault();
+          if (checkedTerms) {
+            submitSignUp();
+          }
+        }}
+      >
+        <label htmlFor="id_terms">
+          <input
+            id="id_terms"
+            type="checkbox"
+            name="terms"
+            required
+            checked={checkedTerms}
+            onChange={(event) => {
+              setCheckedTerms(event.target.checked);
+            }}
+          />{" "}
+          I agree to Mozilla's{" "}
+          <a
+            href="https://www.mozilla.org/about/legal/terms/mozilla"
+            target="_blank"
+            rel="noreferrer noopener"
+          >
+            Terms
+          </a>{" "}
+          and{" "}
+          <a
+            href="https://www.mozilla.org/privacy/websites/"
+            target="_blank"
+            rel="noreferrer noopener"
+          >
+            Privacy Notice
+          </a>
+          .
+        </label>
 
-      <button type="submit" className="button" disabled={!checkedTerms}>
-        Create account
-      </button>
-    </form>
+        <button
+          type="submit"
+          className={!checkedTerms ? "button inactive" : "button"}
+          disabled={!checkedTerms}
+        >
+          Complete sign-in
+        </button>
+      </form>
+    </>
   );
 }
 
-function DisplaySignupProvider({ provider }: { provider: string }) {
-  if (!provider) {
-    // Exit early because there's nothing useful we can say
-    return null;
-  }
+function DisplaySignupDetails({
+  provider,
+  username,
+}: {
+  provider: string;
+  username: string;
+}) {
   let providerVerbose = provider.charAt(0).toUpperCase() + provider.slice(1);
   if (provider === "github") {
     providerVerbose = "GitHub";
   }
   return (
-    <p>
-      You are signing in to MDN Web Docs with <b>{providerVerbose}</b>.
+    <p className="lead">
+      You are signing in to MDN Web Docs with{" "}
+      <span className="provider-name">{providerVerbose}</span>
```

Comment:
Why can't that just be a regular `<b>` (or `<strong>`)? Is that causing problems for screen readers?

Reply:
Nope, I simply prefer having this hook should we want to do more in the future. Perhaps add an icon, different styling for the text, who knows. I prefer to generally keep presentational markup our of HTML and keep it in CSS

Reply:
Interesting! Thanks for sharing. 

But I wonder, if you one day decide to add an icon to that or something, can't you reach that from `.lead b`? Or can't you make the conversion from a simple semantic HTML to something more complex *that* day? You'd have to edit the file in the same directory anyway?

It just feels so overly complicated to turn a perfect simple `b` into `span className="provider-name"` plus all the CSS that has to go with it.